src/utils/dbConnection.js

```diff
@@ -0,0 +1,34 @@
+const mongoose = require('mongoose');
+
+const dbConnection = (function () {
+	const url = process.env.MONGO_DB_URL || 'mongodb://localhost:27017';
+	let mongoClient; 
+
+	async function dbConnect() {
+		mongoClient = await mongoose.connect(url, {
+			useNewUrlParser: true,
+			poolSize: 5,
+		});
+		return mongoClient;
+	}
+
+	async function getdbClient() {
+		try {
+			if (mongoClient) {
+				console.log('db connection is already alive');
+				return mongoClient;
+			} 
+			console.log('Establishing new connection');
+			mongoClient = await dbConnect();
```

Comment:
if we are making the singleton connection then what is the use of pool then

Reply:
I think when we will get instance of db .. mongoose driver will have array of connection i.e. poolSize =5 . I think we are making it complex let's OOB event provided by mongoose.